services/docsrs/docsrs.service.js

```diff
@@ -0,0 +1,53 @@
+'use strict'
+
+const Joi = require('joi')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  build_status: Joi.boolean().required(),
+}).required()
+
+module.exports = class DocsRs extends BaseJsonService {
+  static category = 'build'
+  static route = { base: 'docsrs', pattern: ':crate/:version?' }
+  static examples = [
+    {
+      title: 'docs.rs',
+      namedParams: { crate: 'regex', version: 'latest' },
+      staticPreview: this.render({ version: 'latest', buildStatus: true }),
+      keywords: ['rust'],
+    },
+  ]
+
+  static defaultBadgeData = { label: 'docs' }
+
+  static render({ buildStatus, version }) {
+    if (buildStatus) {
+      return {
+        message: version,
+        color: 'success',
+      }
+    } else {
+      return {
+        message: `${version} | failed`,
+        color: 'critical',
+        isError: true,
+      }
+    }
+  }
```

Comment:
I think we'll need to go in a different direction in terms of the badge content. One of the core goals of Shields is to provide consistency across our badge offering, and sometimes that means our badges look different than the badges provided directly by the upstream services.

Specifically, the badge message would need to reflect the documentation build status, like our other build status badges do (`passing` or `failing`). If there's a desire to include the crate version as a data point within the badge, then that should go in the label following our standard conventions. I know this will result in a different looking badge than what Docs.rs natively provides, but it would be much more consistent with our badges.

For example, crates download count for a specific version
https://img.shields.io/crates/dv/regex/1.4.1
![](https://img.shields.io/crates/dv/regex/1.4.1)

vs. latest
https://img.shields.io/crates/dv/regex
![](https://img.shields.io/crates/dv/regex)

vs. total
https://img.shields.io/crates/d/regex
![](https://img.shields.io/crates/d/regex)

@paulmelnikow - you'd suggested some alternative badge content in the linked issue, any thoughts?